firmware/robot2015/cpu/ball_sense.hpp

```diff
@@ -0,0 +1,38 @@
+// Determines if the emitter to reciever beam is broken
+//  while accounting for ambiant light
+#include "mbed.h"
+
+class ball_sense {
+public:
+    ball_sense();
+
+    // Call once per cycle
+    void update_ball_sensor();
+
+    // Returns true if beam is broken (Ball is contained)
+    bool have_ball();
+
+private:
+    DigitalOut* emitter_pin;
+    AnalogIn* detector_pin;
+
+    // Holds the light sensed when
+    //  emitter is lit (sense_light) and
+    //  dark (sense_dark)
+    int sense_light = 0, sense_dark = 0;
```

Comment:
Oops, I meant to just comment on the `sense_light` variable, didn't notice they were on the same line.  I think it's actually fine as it is now.